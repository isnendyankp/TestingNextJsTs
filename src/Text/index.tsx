import { ReactNode } from "react";

interface Props {
    className?: string;
    children: ReactNode;
}