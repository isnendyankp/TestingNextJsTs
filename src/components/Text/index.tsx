import { ReactNode } from "react";
import React from 'react';

interface Props {
    className?: string;
    children: ReactNode;
}

const Text = ({ className, children }: Props) => {
    return (
        <p className={className}>{children}</p>
    )
};

export default Text;