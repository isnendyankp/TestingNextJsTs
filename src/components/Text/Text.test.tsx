import { render, screen } from '@testing-library/react';
import { describe } from 'node:test';
import Text from '.';
import React from 'react';

describe('Unit test for Text Component', () => {
   
    test('component should be return text', () => {
        render (<Text>{'testing text'}</Text>);
        const element = screen.getByText('testing text');
        expect(element).toMatchSnapshot();
    });

    test('component should be return text with className', () => {
        const document = render (<Text className={'testing-class'}>{'testing text'}</Text>);
        expect(document).toMatchSnapshot();
    });
});