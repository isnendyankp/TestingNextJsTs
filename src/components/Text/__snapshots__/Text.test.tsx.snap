// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should be return text 1`] = `
<p>
  testing text
</p>
`;

exports[`component should be return text with className 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="testing-class"
      >
        testing text
      </p>
    </div>
  </body>,
  "container": <div>
    <p
      class="testing-class"
    >
      testing text
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
